packages/whenable/src/watch.js

```diff
@@ -0,0 +1,193 @@
+// @ts-check
+/* global globalThis */
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+
+import { prepareWhenableKits } from './whenable.js';
+import { getWhenable0, getFirstWhenable } from './whenable-utils.js';
+
+const { Fail } = assert;
+
+export const PromiseWatcherI = M.interface('PromiseWatcher', {
+  onFulfilled: M.call(M.any()).rest(M.any()).returns(M.any()),
+  onRejected: M.call(M.any()).rest(M.any()).returns(M.any()),
+});
+
+/**
+ * @typedef {object} PromiseWatcher
+ * @property {(...args: unknown[]) => void} [onFulfilled]
+ * @property {(...args: unknown[]) => void} [onRejected]
+ */
+
+/** @type {(p: PromiseLike<any>, watcher: PromiseWatcher, ...args: unknown[]) => void} */
+let watchPromise = /** @type {any} */ (globalThis).VatData?.watchPromise;
+if (!watchPromise) {
+  /**
+   * Adapt a promise watcher method to E.when.
+   * @param {Record<PropertyKey, (...args: unknown[]) => unknown>} that
+   * @param {PropertyKey} prop
+   * @param {unknown[]} postArgs
+   */
+  const callMeMaybe = (that, prop, postArgs) => {
+    const fn = that[prop];
+    if (typeof fn !== 'function') {
+      return undefined;
+    }
+    /**
+     * @param {unknown} arg value or reason
+     */
+    const wrapped = arg => {
+      // Don't return a value, to prevent E.when from subscribing to a resulting
+      // promise.
+      fn.call(that, arg, ...postArgs);
+    };
+    return wrapped;
+  };
+
+  // Shim the promise watcher behaviour when VatData.watchPromise is not available.
+  watchPromise = (p, watcher, ...args) => {
+    const onFulfilled = callMeMaybe(watcher, 'onFulfilled', args);
+    const onRejected = callMeMaybe(watcher, 'onRejected', args);
+    onFulfilled ||
+      onRejected ||
+      Fail`promise watcher must implement at least one handler method`;
+    void E.when(p, onFulfilled, onRejected);
+  };
+}
+
+/**
+ * @param {any} specimen
+ * @param {import('./types.js').Watcher} watcher
+ */
+const watchWhenable = (specimen, watcher) => {
```

Comment:
Your review comment says
`watch(specimen, { onFulfillment }, { onRejection }): Whenable<T>`
Did you mean
`watch(specimen, { onFulfillment, onRejection }): Whenable<T>`
?

Reply:
Good catch.  Updated the comment.